src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DateTime.cs

```diff
@@ -264,6 +285,18 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         public override IEnumerable<string> GetRequiredEnumNames()
         {
             return new List<string>() { LanguageConstants.StartOfWeekEnumString };
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            var fValid = base.CheckTypes(context, args, argTypes, errors, out _, out nodeToCoercedTypeMap);
+
+            // As this is an integer returning function, it can be either a number or a decimal depending on NumberIsFloat.
+            // We do this to preserve decimal precision if this function is used in a calculation
+            // since returning Float would promote everything to Float and precision could be lost
+            returnType = context.NumberIsFloat ? DType.Number : DType.Decimal;

```

Comment:
>context.NumberIsFloat ? DType.Number : DType.Decimal; [](http://example.com/codeflow?start=25&length=53)

As this is used in MANY locations, wondering if that couldn't be useful to have a method like
context.GetDefaultNumericType()

Reply:
I hear you, but if the check is as simple as this one, I prefer to keep it exposed so that the underlying logic could be seen.  If it was anything more than a single ternary operator, I'd put it in a function.